ocaml/typing/predef.ml

```diff
@@ -460,6 +472,19 @@ let add_small_number_extension_types add_type env =
        ~jkind:(Jkind.Primitive.float32 ~why:(Primitive ident_unboxed_float32))
        ~jkind_annotation:Jkind.Const.Primitive.float32
 
+let add_or_null add_type env =
+  let add_type1 = mk_add_type1 add_type in
+  env
+  |> add_type1 ident_or_null
+  ~variance:Variance.covariant
+  ~separability:Separability.Sep
```

Comment:
But this is a lie: this type is *not* separable. We've been thinking this is OK because we prevent `or_null`s from going into arrays. And I can't think of a problem here. But there should be a loud comment (and perhaps some discussion) about this.

Maybe, really, in our new world, we have the property that a type is separable if the type system thinks it is separable according to the old analysis _and_ the type is `non_null`. That's probably right.

Reply:
Added a comment.